Development handle forget password

Omar622 opened the pull request:
## Description
- Add 3 endpoints
  - `/user/forget_password` a POST restful API to send email with code to the user
  - `/user/forget_password/verify`  a POST restful API to verify code and return token if code verified
  - `/user/password`  a PUT restful API to update password (have auth)
- Add tests for the two endpoints
 
## Issues
- [handle forget password](https://github.com/codescalersinternships/Flyspray/issues/56)

Omar622 committed:
Add forget password email sender

- fix `email-sender.go` name

Omar622 committed:
Add forget password handler & its tests

Omar622 committed:
Add verify forget password handler and its tests

- update DB `UpdateUser` method

Omar622 committed:
Format code

rawdaGastan commented on backend/app/auth_handlers.go:
```diff
@@ -323,3 +336,94 @@ func (a *App) refreshToken(ctx *gin.Context) (interface{}, Response) {
+	return ResponseMsg{Message: "forget password code has been sent to your email"}, Ok()
+}
+
+func (a *App) verifyForgetPassword(ctx *gin.Context) (interface{}, Response) {
```
here we should have 2 end points
- to verify the code (send a token to the user to be able to call change password)
- to change the password (must have auth)

rawdaGastan reviewed (changes requested):
please add your endpoint to swagger

Omar622 committed:
Merge branch 'development' into development_handle_forget_password

Omar622 committed:
Merge branch 'development' into development_handle_forget_password

Omar622 committed:
Add change password endpoint & its tests

- edit verify forget password endpoint
- update tests

Omar622 committed:
Update swagger docs

- add `forgetPassword` docs
- add `verifyForgetPassword` docs
- add `changePassword` docs

rawdaGastan commented on backend/app/auth_handlers.go:
```diff
@@ -29,8 +29,17 @@ type signinBody struct {
 }
 
 type verifyBody struct {
```
please make them `body` or `input` as both are used

rawdaGastan commented on backend/app/auth_handlers.go:
```diff
@@ -29,8 +29,17 @@ type signinBody struct {
+	Email            string `json:"email" binding:"required"`
+}
+
+type emailInput struct {
```
what is the email input?

rawdaGastan commented on backend/app/auth_handlers.go:
```diff
@@ -403,3 +414,158 @@ func (a *App) refreshToken(ctx *gin.Context) (interface{}, Response) {
+// @Failure 500 {object} Response "Internal server error"
+// @Router /user/forget_password [post]
+func (a *App) forgetPassword(ctx *gin.Context) (interface{}, Response) {
+	var input emailInput
```
include inputs in swagger
check all

Omar622 committed:
Enhance variable name

- change `emailInput` to `forgetPasswordBody`

Omar622 committed:
Include endpoints' input in swagger

Omar622 committed:
Merge branch 'development' into development_handle_forget_password
